src/crimson/osd/heartbeat.cc

```diff
@@ -184,7 +184,7 @@ seastar::future<> Heartbeat::update_peers(int whoami)
     vector<int> want;
     auto osdmap = service.get_osdmap_service().get_map();
     for (auto next = osdmap->get_next_up_osd_after(whoami);
-      peers.size() < min_peers && next >= 0 && next != whoami;
+      peers.size() + want.size() < min_peers && next >= 0 && next != whoami;
```

Comment:
instead of using `want` for collecting the new peers, can we just put
```c++
add_peer(next, osdmap->get_epoch());
```
in the for loop?

Reply:
fixed